src/unicorn/optimize.rs

```diff
@@ -339,6 +383,26 @@ impl<'a, S: SMTSolver> ConstantFolder<'a, S> {
         self.fold_any_binary(left, right, |a, b| a & b, "AND")
     }
 
+    fn fold_or(&self, left: &NodeRef, right: &NodeRef, sort: &NodeType) -> Option<NodeRef> {
+        if is_const_zeroes(left, sort) {
+            trace!("Folding OR(zeroes,x) -> x");
+            return Some(right.clone());
+        }
+        if is_const_zeroes(right, sort) {
+            trace!("Folding OR(x,zeroes) -> x");
+            return Some(left.clone());
+        }
+        if is_const_ones(left, sort) {
+            trace!("Strength-reducing AND(ones,_) -> ones");
```

Comment:
nit-pick: The tracing output should say "OR(...)" instead.

Reply:
done!